### PR TITLE
update config.toml for 1.23 release

### PR DESCRIPTION
**Pull request by jlbutler**
Updated config.toml for the dev-1.23 branch to prepare for the v1.23 release

Patch versions were updated using [patch-releases.md](https://github.com/kubernetes/website/blob/main/content/en/releases/patch-releases.md)

/cc @jimangel @reylejano 

/assign @sftim

### CODE REVIEW

File: config.toml [hunk 1]
```diff
@@ -205,17 +212,12 @@ githubbranch = "v1.19.14"
...
-githubbranch = "v1.18.20"
-docsbranch = "release-1.18"
-url = "https://v1-18.docs.kubernetes.io"
-
 # User interface configuration
 [params.ui]
 # Enable to show the side bar menu in its compact state.
 sidebar_menu_compact = false
+# https://github.com/gohugoio/hugo/issues/8918#issuecomment-903314696
+sidebar_cache_limit = 1
```
Comment (sftim) [inline config.toml:220]: I would omit this change from _this_ PR and merge it in separately.

**Review by sftim — DISMISSED** [review]
One (important) nit.

File: config.toml (hunk 1 above)
Comment (jlbutler) [reply config.toml:220]: yep thanks for the catch!

**Review by sftim — COMMENTED** [review]
/approve

**Comment by reylejano** [comment]
Thank you for this PR
/lgtm

**Pull request merged**